src/alloc_bytes.rs

```diff
@@ -58,25 +53,23 @@ impl std::ops::DerefMut for MiriAllocBytes {
 impl MiriAllocBytes {
     /// This method factors out how a `MiriAllocBytes` object is allocated,
     /// specifically given an allocation function `alloc_fn`.
-    /// `alloc_fn` is only used if `size != 0`.
-    /// Returns `Err(layout)` if the allocation function returns a `ptr` that is `ptr.is_null()`.
+    /// `alloc_fn` is only used with `size != 0`.
+    /// Returns `Err(layout)` if the allocation function returns a `ptr` where `ptr.is_null()`.
     fn alloc_with(
         size: usize,
         align: usize,
         alloc_fn: impl FnOnce(Layout) -> *mut u8,
     ) -> Result<MiriAllocBytes, Layout> {
+        // When size is 0 we allocate 1 byte anyway, so addresses don't possibly overlap.
```

Comment:
```suggestion
        // When size is 0 we allocate 1 byte anyway, to ensure each allocation has a unique address.
```

Reply:
I had to change some things around but the relevant part of this comment is in the function documentation now.